rancher2/structure_cluster_rke_config_services_kube_api.go

```diff
@@ -210,6 +263,55 @@ func expandClusterRKEConfigServicesKubeAPIAuditLog(p []interface{}) (*management
 	return obj, nil
 }
 
+func expandClusterRKEConfigServicesKubeAPIAdmissionConfigurationPlugins(p []interface{}) ([]interface{}, error) {
+	obj := []interface{}{}
+	if len(p) == 0 || p[0] == nil {
+		return obj, nil
+	}
+	for _, plugin := range p {
+		pluginMap, ok := plugin.(map[string]interface{})
+		if !ok || len(pluginMap) == 0 {
+			continue // or return, if this is an error condition
+		}
+		var newObj = map[string]interface{}{}
+		newObj["name"] = pluginMap["name"].(string)
+		newObj["path"] = pluginMap["path"].(string)
+		configuration, err := ghodssyamlToMapInterface(pluginMap["configuration"].(string))
+		if err != nil {
+			return obj, fmt.Errorf("Unmarshalling plugin configuration from yaml for %v: %v", pluginMap["name"], err)
+		}
+		newObj["configuration"] = configuration
+		obj = append(obj, newObj)
+	}
+	return obj, nil
+}
+
+func expandClusterRKEConfigServicesKubeAPIAdmissionConfiguration(p []interface{}) (map[string]interface{}, error) {
```

Comment:
The code you've highlighted with this comment is not the migration logic from what I can tell, it looks like the migration logic added in this PR is actually in the `rancher2/resource_rancher2_cluster.go` file on line 107. 

This code looks the same because this PR is attempting to introduce the same schema as the prior PR, but with migration logic included to move the old schema into the new schema. The core principal of this PR is a schema change so that admission configurations can be used more easily / at all. 

I agree, this shouldn't be merged until that migration path can be properly tested. Currently the PR states `Was not able to test on a cluster with an existing admission configuration imported by the previous tfp version.` which leaves room for regressions.

Reply:
After offline discussion, I agree and will request testing for the migration state logic.

Reply:
Tested without provisioning a cluster with admission_config enabled, as discussed (see latest comments for update).